src/Features/PaymentGatewaySuggestions/EvaluateSuggestion.php

```diff
@@ -16,12 +16,12 @@ class EvaluateSuggestion {
 	/**
 	 * Evaluates the spec and returns the suggestion.
 	 *
-	 * @param array $spec The suggestion to evaluate.
-	 * @return array The evaluated suggestion.
+	 * @param object|array $spec The suggestion to evaluate.
+	 * @return object The evaluated suggestion.
 	 */
 	public static function evaluate( $spec ) {
 		$rule_evaluator = new RuleEvaluator();
-		$suggestion     = (object) $spec;
+		$suggestion     = is_array( $spec ) ? (object) $spec : clone $spec;
```

Comment:
I can't remember if this works in PHP or not, but can we clone and coerce to an object at the same time?

```suggestion
		$suggestion     =  clone (object) $spec;
```

Reply:
Yeah we could do that, this would technically cause a array to be cloned twice (in essence), creating a new object out of it first and then cloning that new object.

Reply:
Ah, good point.  Let's leave it as is.